realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -437,7 +442,16 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
                 // refresh the Realm to the version provided by the worker thread
                 // (advanceRead to the latest version may cause a version mismatch error) preventing us
                 // from importing correctly the handover table view
-                realm.sharedGroupManager.advanceRead(result.versionID);
+                try {
+                    realm.sharedGroupManager.advanceRead(result.versionID);
+                } catch (BadVersionException e) {
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jcjacksonengineering@gmail.com
    Domain joe-jackson.net has exceeded the max emails per hour (114/100 (113%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext4.iad.github.net ([192.30.252.195]:34864 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by rs5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1afmMk-001xaj-9f
	for jcjackson@joe-jackson.net; Tue, 15 Mar 2016 06:38:18 -0400
Date: Tue, 15 Mar 2016 01:12:16 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1458029536;
	bh=6fmgPSFZh/596954I49I2LVy7j7A/eBP/BKuTLDyPC0=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=OFnJmxxMcZgVUFC5d41Mp2uydnQScSfjucYCUQn+zKIRQLOSJVCGXv7sbwNkPufxO
	 3yjzixwKd9j3iVk3AVamePac9TUCs26YUghrLbnN13//GugUB6t2V/Ji9JaMVYsOfW
	 I5JNiW+lJnuazsRNvpqUXQ/2qH7aEY44nYIlMcEw=
From: Chen Mulong <notifications@github.com>
Reply-To: realm/realm-java <reply@reply.github.com>
To: realm/realm-java <realm-java@noreply.github.com>
Message-ID: <realm/realm-java/pull/2450/r56127126@github.com>
In-Reply-To: <realm/realm-java/pull/2450@github.com>
References: <realm/realm-java/pull/2450@github.com>
Subject: Re: [realm-java] Fix BadVersionException (#2450)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_56e7c3e0646c9_54c53f979ac5729c42991e";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: beeender
X-GitHub-Recipient: jcj52436999
List-ID: realm/realm-java <realm-java.realm.github.com>
List-Archive: https://github.com/realm/realm-java
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+000f94f7ee5059e6b42479cde31aa0383fd17b0a4162f97c92cf0000000112ff85e092a170ce03586e96@reply.github.com>,
 <https://github.com/notifications/unsubscribe/AA-U9yQJUnqgLbIgASXYX_mFbVmwwcpUks5ptmnggaJpZM4HwK21>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: jcjackson@joe-jackson.net


----==_mimepart_56e7c3e0646c9_54c53f979ac5729c42991e
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

> @@ -437,7 +442,16 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
>                  // refresh the Realm to the version provided by the worker thread
>                  // (advanceRead to the latest version may cause a version mismatch error) preventing us
>                  // from importing correctly the handover table view
> -                realm.sharedGroupManager.advanceRead(result.versionID);
> +                try {
> +                    realm.sharedGroupManager.advanceRead(result.versionID);
> +                } catch (BadVersionException e) {

I think it is totally safe to continue using the `SharedGroup` even if the `advance_read` failed because of `BadVersion`, right? @finnschiermer 

---
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/realm/realm-java/pull/2450/files#r56127126
----==_mimepart_56e7c3e0646c9_54c53f979ac5729c42991e
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/realm/realm-java/pull/2450#discussion_r56127126">realm/realm-library/src/main/java/io/realm/HandlerController.java</a>:</p>
<pre style='color:#555'>&gt; @@ -437,7 +442,16 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
&gt;                  // refresh the Realm to the version provided by the worker thread
&gt;                  // (advanceRead to the latest version may cause a version mismatch error) preventing us
&gt;                  // from importing correctly the handover table view
&gt; -                realm.sharedGroupManager.advanceRead(result.versionID);
&gt; +                try {
&gt; +                    realm.sharedGroupManager.advanceRead(result.versionID);
&gt; +                } catch (BadVersionException e) {
</pre>
<p>I think it is totally safe to continue using the <code>SharedGroup</code> even if the <code>advance_read</code> failed because of <code>BadVersion</code>, right? <a href="https://github.com/finnschiermer" class="user-mention">@finnschiermer</a> </p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>You are receiving this because you are subscribed to this thread.<br>Reply to this email directly or view it on GitHub:<br><a href="https://github.com/realm/realm-java/pull/2450/files#r56127126">https://github.com/realm/realm-java/pull/2450/files#r56127126</a><img alt="" height="1" src="https://github.com/notifications/beacon/AA-U91Cok9IqfPAbupwA56AC8TRYUKysks5ptmnggaJpZM4HwK21.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/realm/realm-java/pull/2450/files#r56127126"></link>
  <meta itemprop="name" content="View Pull Request"></meta>
</div>
<meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_56e7c3e0646c9_54c53f979ac5729c42991e--